path/join_test.ts

```diff
@@ -0,0 +1,124 @@
+import { test, assertEqual } from "https://deno.land/x/testing/testing.ts";
+import * as path from "./index";
+
+const backslashRE = /\\/g;
+
+const joinTests =
+  // arguments                     result
+  [
+    [[".", "x/b", "..", "/b/c.js"], "x/b/c.js"],
+    [[], "."],
+    [["/.", "x/b", "..", "/b/c.js"], "/x/b/c.js"],
+    [["/foo", "../../../bar"], "/bar"],
+    [["foo", "../../../bar"], "../../bar"],
+    [["foo/", "../../../bar"], "../../bar"],
+    [["foo/x", "../../../bar"], "../bar"],
+    [["foo/x", "./bar"], "foo/x/bar"],
+    [["foo/x/", "./bar"], "foo/x/bar"],
+    [["foo/x/", ".", "bar"], "foo/x/bar"],
+    [["./"], "./"],
+    [[".", "./"], "./"],
+    [[".", ".", "."], "."],
+    [[".", "./", "."], "."],
+    [[".", "/./", "."], "."],
+    [[".", "/////./", "."], "."],
+    [["."], "."],
+    [["", "."], "."],
+    [["", "foo"], "foo"],
+    [["foo", "/bar"], "foo/bar"],
+    [["", "/foo"], "/foo"],
+    [["", "", "/foo"], "/foo"],
+    [["", "", "foo"], "foo"],
+    [["foo", ""], "foo"],
+    [["foo/", ""], "foo/"],
+    [["foo", "", "/bar"], "foo/bar"],
+    [["./", "..", "/foo"], "../foo"],
+    [["./", "..", "..", "/foo"], "../../foo"],
+    [[".", "..", "..", "/foo"], "../../foo"],
+    [["", "..", "..", "/foo"], "../../foo"],
+    [["/"], "/"],
+    [["/", "."], "/"],
+    [["/", ".."], "/"],
+    [["/", "..", ".."], "/"],
+    [[""], "."],
+    [["", ""], "."],
+    [[" /foo"], " /foo"],
+    [[" ", "foo"], " /foo"],
+    [[" ", "."], " "],
+    [[" ", "/"], " /"],
+    [[" ", ""], " "],
+    [["/", "foo"], "/foo"],
+    [["/", "/foo"], "/foo"],
+    [["/", "//foo"], "/foo"],
+    [["/", "", "/foo"], "/foo"],
+    [["", "/", "foo"], "/foo"],
+    [["", "/", "/foo"], "/foo"]
```

Comment:
Are these tests lifted from somewhere? If so we need to cite them.

Reply:
These tests are referenced from https://github.com/browserify/path-browserify/blob/master/test/

Reply:
Please add a header to each file that originated from there (all of them?) 
```
// Copyright the Browserify authors. MIT License.
// Ported from https://github.com/browserify/path-browserify/
```

Reply:
Copyright belongs to Node contributes.